source/ui/BioXAS/BioXASSSRLMonochromatorRegionControlView.cpp

```diff
@@ -0,0 +1,177 @@
+#include "BioXASSSRLMonochromatorRegionControlView.h"
+#include "beamline/BioXAS/BioXASSSRLMonochromator.h"
+#include "beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.h"
+
+BioXASSSRLMonochromatorRegionControlView::BioXASSSRLMonochromatorRegionControlView(BioXASSSRLMonochromatorRegionControl *regionControl, QWidget *parent) :
+	QGroupBox(parent)
+{
+	// Initialize member variables.
+
+	regionControl_ = 0;
+
+	// Create UI elements.
+
+	QLabel *slitsStatusPrompt = new QLabel("Slits status:");
+	slitsStatusGreen_ = new QLabel();
+	slitsStatusRed_ = new QLabel();
+
+	QLabel *paddleStatusPrompt = new QLabel ("Paddle status:");
+	paddleStatusGreen_ = new QLabel();
+	paddleStatusRed_ = new QLabel();
+
+	QLabel *keyStatusPrompt = new QLabel("Key status:");
+	keyStatusGreen_ = new QLabel();
+	keyStatusRed_ = new QLabel();
+
+	QLabel *brakeStatusPrompt = new QLabel("Brake status:");
+	brakeStatusGreen_ = new QLabel();
+	brakeStatusRed_ = new QLabel();
+
+	// Create and set layouts.
+
+	QGridLayout *statusLayout = new QGridLayout();
+	statusLayout->addWidget(slitsStatusPrompt, 0, 0, 1, 1, Qt::AlignRight);
+	statusLayout->addWidget(slitsStatusGreen_, 0, 1, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(slitsStatusRed_, 0, 2, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(paddleStatusPrompt, 1, 0, 1, 1, Qt::AlignRight);
+	statusLayout->addWidget(paddleStatusGreen_, 1, 1, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(paddleStatusRed_, 1, 2, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(keyStatusPrompt, 2, 0, 1, 1, Qt::AlignRight);
+	statusLayout->addWidget(keyStatusGreen_, 2, 1, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(keyStatusRed_, 2, 2, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(brakeStatusPrompt, 3, 0, 1, 1, Qt::AlignRight);
+	statusLayout->addWidget(brakeStatusGreen_, 3, 1, 1, 1, Qt::AlignCenter);
+	statusLayout->addWidget(brakeStatusRed_, 3, 2, 1, 1, Qt::AlignCenter);
+
+	setLayout(statusLayout);
+	setFlat(true);
+
+	// Initial settings.
+
+	slitsStatusGreen_->setPixmap(QPixmap(":/22x22/greenLEDOff.png"));
+	slitsStatusRed_->setPixmap(QPixmap(":/22x22/redLEDOff.png"));
+
+	paddleStatusGreen_->setPixmap(QPixmap(":/22x22/greenLEDOff.png"));
+	paddleStatusRed_->setPixmap(QPixmap(":/22x22/redLEDOff.png"));
+
+	keyStatusGreen_->setPixmap(QPixmap(":/22x22/greenLEDOff.png"));
+	keyStatusRed_->setPixmap(QPixmap(":/22x22/redLEDOff.png"));
+
+	brakeStatusGreen_->setPixmap(QPixmap(":/22x22/greenLEDOff.png"));
+	brakeStatusRed_->setPixmap(QPixmap(":/22x22/redLEDOff.png"));
+
+	// Current settings.
+
+	setRegionControl(regionControl);
+}
+
+BioXASSSRLMonochromatorRegionControlView::~BioXASSSRLMonochromatorRegionControlView()
+{
+
+}
+
+void BioXASSSRLMonochromatorRegionControlView::setRegionControl(BioXASSSRLMonochromatorRegionControl *newControl)
+{
+	if (regionControl_ != newControl) {
+
+		if (regionControl_)
```

Comment:
See my earlier comment about the setXYZControl methods.  This is halfway there though.  It disconnects the old control from object, but doesn't handle the ownership issue.